beacon-chain/state/state-native/state_trie.go

```diff
@@ -490,6 +507,11 @@ func (b *BeaconState) Copy() state.BeaconState {
 		fieldCount = params.BeaconConfig().BeaconStateCapellaFieldCount
 	}
 
+	// TODO: may panic!
+	header, err := b.latestExecutionPayloadHeaderVal()
+	if err != nil {
+		panic(err)
```

Comment:
Copying the beacon state may theoretically fail but in reality cannot unless we introduce it's been fitted with an unsupported execution payload header.

I didn't want to update the `Copy` interface to also return an error because this will affect the interfaces of much more functions throughout the codebase (including the `forkchoice.HeadState()`). It's not possible for user input to trigger a panic here, and with sufficient test coverage, I don't expect devs to introduce unsupported execution payloads here. lmkwyt!